src/typeahead/typeahead.js

```diff
@@ -529,6 +545,9 @@ angular.module('ui.bootstrap.typeahead', ['ui.bootstrap.debounce', 'ui.bootstrap
           isLoadingSetter(originalScope, false);
           cancelPreviousTimeout();
           resetMatches();
+          if(!inputValue) {
+            scope.selectedM = false;//hide clear button if input empty
```

Comment:
please see comments above about comment formatting.  also leave a space between the `;` and the `//`.